### DISCUSSION TITLE
LWM2M Interface Websocket Server

### DISCUSSION THREAD

**Discussion by RIPHeisenberg** [zephyrproject-rtos/zephyr#77984] [Q&A] [answered]
Hi everyone,

I am currently developing an HTTP server to configure some data for my STM32. We want to be able to use DHCP for this. However, the issue is that the Zephyr Web API only supports static IP configurations (see samples/net/sockets/http_server).

Is there a possibility to configure the IP address at runtime?

Any help would be greatly appreciated.

Best regards

**Comment by jukkar** [comment]
Hi @RIPHeisenberg, we should allow NULL when defining the service. It seems to work after some tweaks, I will send a PR for this.

**Reply by RIPHeisenberg** [reply]
Hi @jukkar thanks for considering this feature. I think the Implementation is very well done in general

**Comment by jukkar** [comment]
As a workaround, you can also specify "0.0.0.0" as a host address.

**Reply by RIPHeisenberg** [reply]
Thanks I'll try it out!

**Comment by jukkar** [comment, marked as answer]
I sent a PR for this too https://github.com/zephyrproject-rtos/zephyr/pull/78001

**Comment by jukkar** [comment]
Closing this as the functionality has been implemented.
